packages/react-inbox/src/components/Message/index.tsx

```diff
@@ -58,44 +61,35 @@ const Message: React.FunctionComponent<MessageProps> = ({
     });
   }, [created]);
 
-  const actions = useMemo(() => {
-    return [{ href: data?.clickAction, label: "View Details" }];
-  }, [data]);
-
+  const actions = useMemo(() => [{
+    href: data?.clickAction,
+    label: "View Details",
+  }], [data]);
+  const showMarkAsRead = !read && readTrackingId;
+  const showMarkAsUnread = read && unreadTrackingId;
   const options = useMemo(
     () =>
       [
-        !read &&
-          trackingIds?.readTrackingId && {
-            label: "Mark as Read",
-            onClick: () => {
-              if (!trackingIds?.readTrackingId) {
-                return;
-              }
-
-              markMessageRead(messageId, trackingIds?.readTrackingId);
-            },
+        showMarkAsRead && {
+          label: "Mark as Read",
```

Comment:
are we sure okay to be hard coding `Mark as Read` string?

Reply:
Fixed.